lib/kafo/migrations.rb

```diff
@@ -0,0 +1,56 @@
+require 'yaml'
+require 'kafo/migration_context'
+
+module Kafo
+  class Migrations
+
+    attr_reader :processed
+
+    def initialize(migrations_dir)
+      @migrations_dir = migrations_dir
+      @migrations = {}
+      @processed = 0
+      @applied_file = File.join(@migrations_dir, '.applied')
+    end
+
+    def applied
+      @applied ||= load_applied
+    end
+
+    def load_migrations
+      Dir.glob(@migrations_dir + "/*.rb").each do |file|
+        next if applied.include?(File.basename(file))
+        KafoConfigure.logger.debug "Loading migration #{file}"
+        migration = File.read(file)
+        migration_block = proc { instance_eval(migration, file, 1) }
+        add_migration(file, &migration_block)
+      end
+    end
+
+    def add_migration(name, &block)
+      @migrations[name] = block
+    end
+
+    def run(scenario, answers)
+      @processed = 0
+      @migrations.keys.sort.each do |name|
+        KafoConfigure.logger.debug "Executing migration #{name}"
+        migration = @migrations[name]
+        scenario, answers = Kafo::MigrationContext.execute(scenario, answers, &migration)
+        @processed += 1
```

Comment:
is @processed == @migrations.size? all loaded migrations will be executed here I think